Automattic/sensei
# Use student progress models on the frontend

Resolves part of #7150 

- Here I also changed logic for table reading quiz progress repository, to create lesson progress if it doesn't exist yet (creating a quiz progress leads to an error).
- In student progress migration, I added handling for the failed status. It is not very efficient as it performs a query in the loop. Any ideas on how it improve?
- When table reading is enabled, tests fail: we need to update code for quiz submission, then tests themselves to make them green.

## Proposed Changes
* Add a `find` method for progress repositories to get multiple progress entries.
* Adjust client code on the frontend to use new progress storage instead of activity comments.
* Adjust client code to use separated lesson/quiz progress behavior.

## Testing Instructions

1. Create a course with a few lessons with quizzes.
2. Create a few students.
3. Without enabling the feature, complete a course. Make sure frontend looks good (correctly displays the progress), you can see submissions and try to grade them.
4. Enable the tables-based progress feature, complete a course. Make sure frontend looks good (correctly displays the progress), you can see submissions and try to grade them.
5. Run migration.
6. Enable reading from tables. Make sure frontend looks good (correctly displays the progress), you can see submissions and try to grade them.
7. Reset progress. Make sure you don't see progress on the frontend.

## New/Updated Hooks


*

## Deprecated Code


*

## Pre-Merge Checklist
